chirp/drivers/tk11.py

```diff
@@ -0,0 +1,3297 @@
+    u8 dual_watch;
+    u8 backlight;
+    ul16 call_ch;
+    u8 beep;
+    u8 key_short1;
+    u8 key_long1;
+    u8 key_short2;
+    u8 key_long2;
+    u8 scan_mode;
+    u8 auto_lock;
+    u8 power_on_screen_mode;
+    u8 alarm_mode;
+    u8 roger_tone;
+    u8 repeater_tail;
+    u8 tail_tone;
+    u8 denoise_sw;
+    u8 denoise_lvl;
+    u8 transpositional_sw;
+    u8 transpositional_lvl;
+    u8 chn_A_volume;
+    u8 chn_B_volume;
+    u8 key_tone_flag;
+    u8 language;
+    u8 noaa_same_decode;
+    u8 noaa_same_event;
+    u8 noaa_same_address;
+    u8 unknown;
+    u8 sbar;
+    u8 brightness;
+    u8 kill_code;
+    u8 dtmf_side_tone;
+    u8 dtmf_decode_rspn;
+    u8 match_tot;
+    u8 match_qt_mode;
+    u8 match_dcs_bit;
+    u8 match_threshold;
+    u8 unknown_2;
+    ul16 cw_pitch_freq;
+    u8 unknown_3[4];
+    ul16 dtmf_separator;
+    ul16 dtmf_group_code;
+    u8 dtmf_reset_time;
+    u8 dtmf_resv4;
+    ul16 dtmf_carry_time;
+    ul16 dtmf_first_code_time;
+    ul16 dtmf_d_code_time;
+    ul16 dtmf_continue_time;
+    ul16 dtmf_interval_time;
+    u8 dtmf_id[8];
+    u8 dtmf_up_code[16];
+    u8 dtmf_down_code[16];
+    u8 unknown_4[16];
+    ul16 tone5_separator;
+    ul16 tone5_group_code;
+    u8 tone5_reset_time;
+    u8 tone5_resv4;
+    ul16 tone5_carry_time;
+    ul16 tone5_first_code_time;
+    u8 tone5_protocol;
+    u8 tone5_resv1;
+    ul16 tone5_single_continue_time;
+    ul16 tone5_single_interval_time;
+    u8 tone5_id[8];
+    u8 tone5_up_code[16];
+    u8 tone5_down_code[16];
+    ul16 tone5_user_freq[15];
+    u8 tone5_revs5[2];
+    char logo_string1[16];
+    char logo_string2[16];
+
+    // General 2 collapsed to general
+    u8 unknown_5[48];
+    u8 dtmf_kill[8];
+    u8 dtmf_wakeup[8];
+    u8 tone5_kill[8];
+    u8 tone5_wakeup[8];
+    u8 unknown_6[16];
+    u8 device_name[16];
+} general;
+#printoffset "After general";
+
+struct {
+    ul16 a_id;
+    ul16 b_id;
+    ul16 freq_a_id;
+    ul16 freq_b_id;
+    ul16 channel_a_id;
+    ul16 channel_b_id;
+    ul16 noaa_a_id;
+    ul16 noaa_b_id;
+} channels_idx;
+#printoffset "After channels_idx";
+
+struct {
+    ul16 vfo_frequency;
+    u8 channel_id;
+    u8 memory_vfo_flag;
+    u8 unknown[4];
+    ul16 frequencies[32];
+} fm;
+#printoffset "After fm";
+
+struct {
+    u8 name[16];
+    u8 code_id[8];
+} dtmf_contacts[16];
+#printoffset "After dtmf_contacts";
+
+struct {
+    u8 name[16];
+    ul16 prio_1;
+    ul16 prio_2;
+    u8 unknown[4];
+    ul16 channels[48];
+    u8 unknown_2[8];
+} scan_list[32];
+#printoffset "After scan_list";
+
+struct {
+    u8 name[16];
+    u8 code_id[8];
+} tone5_contacts[16];
+#printoffset "After 5TONE contacts";
+
+struct {
+    u8 address[8];
+    u8 info[32];
+} noaa_decode_addresses[16];
+#printoffset "After NOAA Decode";
+
+struct {
+    u8 value;
+} noaa_same_events_control[128];
+#printoffset "After NOAA SAME event control";
+"""
+
+MAX_CHUNK_SIZE = 0x200
+
+MAGIC_CODE = secrets.randbits(32)
```

Comment:
Can you explain this please? Anything random seems like a bad idea for something like this.

Reply:
This would be the first radio I've ever seen that behaves that way. Does it need to be different each time, or will one constant that we always use actually work? If you make that a different code across a single connection does the radio reject the messages with different code?

If either of these are not true, I'd prefer to make this not random. When you're gone, someone trying to diagnose an issue from looking at serial trace files will appreciate having a consistent reference over something that is always different.